ipaserver/plugins/server.py

```diff
@@ -619,7 +619,10 @@ def _remove_server_principal_references(self, master):
             if master in srvlist:
                 srvlist.remove(master)
                 attr = ' '.join(srvlist)
-                ret['defaultServerList'] = attr
+                if len(srvlist) == 0:
```

Comment:
```python
if not srvlist:
    del ret['defaultServerList']
else:
    ret['defaultServerList'] = ' '.join(srvlist)
```
^-- removes the possibly unnecessary `join()` if the list is empty and the condition is a bit more Pythonic I believe :)

Reply:
+1, done